File: src/NetherNet/ErrorCodes.cs
namespace NetherNet;

public static class ErrorCodes
{
    public const int None = 0;
    public const int DestinationNotLoggedIn = 1;
    public const int NegotiationTimeout = 2;
    public const int WrongTransportVersion = 3;
    public const int FailedToCreatePeerConnection = 4;
    public const int Ice = 5;
    public const int ConnectRequest = 6;
    public const int ConnectResponse = 7;
    public const int CandidateAdd = 8;
    public const int InactivityTimeout = 9;
    public const int FailedToCreateOffer = 10;
    public const int FailedToCreateAnswer = 11;
    public const int FailedToSetLocalDescription = 12;
    public const int FailedToSetRemoteDescription = 13;
    public const int NegotiationTimeoutWaitingForResponse = 14;
    public const int NegotiationTimeoutWaitingForAccept = 15;
    public const int IncomingConnectionIgnored = 16;
    public const int SignalingParsingFailure = 17;
    public const int SignalingUnknownError = 18;
    public const int SignalingUnicastMessageDeliveryFailed = 19;
    public const int SignalingBroadcastDeliveryFailed = 20;
    public const int SignalingMessageDeliveryFailed = 21;
    public const int SignalingTurnAuthFailed = 22;
    public const int SignalingFallbackToBestEffortDelivery = 23;
    public const int NoSignalingChannel = 24;
    public const int NotLoggedIn = 25;
    public const int SignalingFailedToSend = 26;
    public const int RelayServerConfigurationResultFailure = 27;
    public const int RelayServerConfigurationResultParsingErrorNoUrls = 28;
    public const int RelayServerConfigurationResultParsingErrorNoCredentials = 29;
    public const int RelayServerConfigurationResultParsingErrorNoServers = 30;
    public const int RelayServerConfigurationResultParsingErrorNoExpiration = 31;
    public const int DataChannelClosed = 32;
    public const int InternalErrorJsonSerialization = 33;
    public const int InvalidArgument = 34;
    public const int GenericFailure = 35;
    public const int FailedToCreateIdentityAssertion = 36;
    public const int IdentityNotAllowed = 37;
}

File: src/error_codes.zig
/// NetherNet signaling error codes.
pub const ErrorCode = enum(u8) {
    none = 0,

    destination_not_logged_in = 1,
    negotiation_timeout = 2,
    wrong_transport_version = 3,
    failed_to_create_peer_connection = 4,
    ice = 5,
    connect_request = 6,
    connect_response = 7,
    candidate_add = 8,
    inactivity_timeout = 9,

    failed_to_create_offer = 10,
    failed_to_create_answer = 11,
    failed_to_set_local_description = 12,
    failed_to_set_remote_description = 13,

    negotiation_timeout_waiting_for_response = 14,
    negotiation_timeout_waiting_for_accept = 15,
    incoming_connection_ignored = 16,

    signaling_parsing_failure = 17,
    signaling_unknown_error = 18,
    signaling_unicast_message_delivery_failed = 19,
    signaling_broadcast_delivery_failed = 20,
    signaling_message_delivery_failed = 21,
    signaling_turn_auth_failed = 22,
    signaling_fallback_to_best_effort_delivery = 23,
    no_signaling_channel = 24,
    not_logged_in = 25,
    signaling_failed_to_send = 26,

    relay_server_configuration_result_failure = 27,
    relay_server_configuration_result_parsing_error_no_urls = 28,
    relay_server_configuration_result_parsing_error_no_credentials = 29,
    relay_server_configuration_result_parsing_error_no_servers = 30,
    relay_server_configuration_result_parsing_error_no_expiration = 31,

    data_channel_closed = 32,
    internal_error_json_serialization = 33,
    invalid_argument = 34,
    generic_failure = 35,
    failed_to_create_identity_assertion = 36,
    identity_not_allowed = 37,
};
